code/lib/csf-tools/src/enrichCsf.ts

```diff
@@ -9,74 +9,127 @@ export interface EnrichCsfOptions {
-        false,
-        true
-      );
+        t.objectExpression([
+          t.objectProperty(t.identifier('source'), t.stringLiteral(source)),
+          t.spreadElement(optionalStorySource),
+        ])
+      )
+    );
+  }
 
-      parameters.push(
-        t.objectProperty(
-          t.identifier('storySource'),
-          t.objectExpression([
-            t.objectProperty(t.identifier('source'), t.stringLiteral(source)),
-            t.spreadElement(optionalStorySource),
-          ])
-        )
-      );
-    }
+  // docs: { description: { story: %%description%% } },
+  if (description) {
+    const optionalDocs = t.optionalMemberExpression(
+      originalParameters,
+      t.identifier('docs'),
+      false,
+      true
+    );
 
-    // docs: { description: { story: %%description%% } },
-    if (description) {
-      const optionalDocs = t.optionalMemberExpression(
-        originalParameters,
-        t.identifier('docs'),
-        false,
-        true
-      );
+    const optionalDescription = t.optionalMemberExpression(
+      optionalDocs,
+      t.identifier('description'),
+      false,
+      true
+    );
 
-      const optionalDescription = t.optionalMemberExpression(
-        optionalDocs,
-        t.identifier('description'),
-        false,
-        true
-      );
+    parameters.push(
+      t.objectProperty(
+        t.identifier('docs'),
+        t.objectExpression([
+          t.spreadElement(optionalDocs),
+          t.objectProperty(
+            t.identifier('description'),
+            t.objectExpression([
+              t.objectProperty(t.identifier('story'), t.stringLiteral(description)),
+              t.spreadElement(optionalDescription),
+            ])
+          ),
+        ])
+      )
+    );
+  }
+  if (parameters.length > 1) {
+    const addParameter = t.expressionStatement(
+      t.assignmentExpression('=', originalParameters, t.objectExpression(parameters))
+    );
+    csf._ast.program.body.push(addParameter);
+  }
+};
 
-      parameters.push(
-        t.objectProperty(
-          t.identifier('docs'),
-          t.objectExpression([
-            t.spreadElement(optionalDocs),
-            t.objectProperty(
-              t.identifier('description'),
-              t.objectExpression([
-                t.objectProperty(t.identifier('story'), t.stringLiteral(description)),
-                t.spreadElement(optionalDescription),
-              ])
-            ),
-          ])
-        )
-      );
+const addComponentDescription = (
+  node: t.ObjectExpression,
+  path: string[],
+  value: t.ObjectProperty
+) => {
+  if (!path.length) {
+    // make this the lowest-priority so that if the user is object-spreading on top of it,
+    // the users' code will "win"
+    const hasExistingComponent = node.properties.find(
+      (p) => t.isObjectProperty(p) && t.isIdentifier(p.key) && p.key.name === 'component'
+    );
+    if (!hasExistingComponent) {
+      node.properties.unshift(value);
     }
```

Comment:
Think we should move the comment down to the `unshift` call.
```suggestion
    const hasExistingComponent = node.properties.find(
      (p) => t.isObjectProperty(p) && t.isIdentifier(p.key) && p.key.name === 'component'
    );
    if (!hasExistingComponent) {
      // make this the lowest-priority so that if the user is object-spreading on top of it,
      // the users' code will "win"
      node.properties.unshift(value);
    }
```

Reply:
It's probably because chromatic is only testing the stories & not the docs page?